src/sink/kinesis.rs

```diff
@@ -14,9 +14,19 @@ use rusoto_kinesis::{KinesisClient, PutRecordsError, PutRecordsInput,
                      PutRecordsOutput, PutRecordsRequestEntry};
 use rusoto_kinesis::Kinesis as RusotoKinesis;
 use sink::Sink;
-use std::sync;
+use std::sync::Arc;
+use std::sync::atomic::{AtomicUsize, Ordering};
 use util::Valve;
 
+lazy_static! {
+    /// Total records emitted.
+    pub static ref KINESIS_PUBLISH_SUCCESS_SUM: Arc<AtomicUsize> = Arc::new(AtomicUsize::new(0));
+    /// Total retryable publication errors.
+    pub static ref KINESIS_PUBLISH_FAILURE_SUM: Arc<AtomicUsize> = Arc::new(AtomicUsize::new(0));
```

Comment:
How are failure and fatal distinct?

Reply:
The comments should capture it.

Failure is retryable, fatal is not.    Fatal signals that Kinesis publication has stopped.